scripts/syntactics/parser/biaffine_parser.py

```diff
@@ -0,0 +1,370 @@
+
+        Parameters
+        ----------
+        name : str
+            parameter name
+        shape : tuple
+            parameter shape
+        init : mxnet.initializer
+            an initializer
+
+        Returns
+        -------
+        mxnet.gluon.parameter
+            a parameter object
+        """
+        p = self.params.get(name, shape=shape, init=init)
+        return p
+
+    def forward(self, word_inputs, tag_inputs, arc_targets=None, rel_targets=None):
+        """Run decoding
+
+        Parameters
+        ----------
+        word_inputs : mxnet.ndarray.NDArray
+            word indices of seq_len x batch_size
+        tag_inputs : mxnet.ndarray.NDArray
+            tag indices of seq_len x batch_size
+        arc_targets : mxnet.ndarray.NDArray
+            gold arc indices of seq_len x batch_size
+        rel_targets : mxnet.ndarray.NDArray
+            gold rel indices of seq_len x batch_size
+        Returns
+        -------
+        tuple
+            (arc_accuracy, rel_accuracy, overall_accuracy, loss) when training, else if given gold target
+        then return arc_accuracy, rel_accuracy, overall_accuracy, outputs, otherwise return outputs, where outputs is a
+        list of (arcs, rels).
+        """
+        is_train = autograd.is_training()
+
+        def flatten_numpy(ndarray):
+            """Flatten nd-array to 1-d column vector
+
+            Parameters
+            ----------
+            ndarray : numpy.ndarray
+                input tensor
+
+            Returns
+            -------
+            numpy.ndarray
+                A column vector
+
+            """
+            return np.reshape(ndarray, (-1,), 'F')
+
+        batch_size = word_inputs.shape[1]
+        seq_len = word_inputs.shape[0]
+        mask = np.greater(word_inputs, self._vocab.ROOT).astype(np.float32)
+        num_tokens = int(np.sum(mask))  # non padding, non root token number
+
+        if is_train or arc_targets is not None:
+            mask_1D = flatten_numpy(mask)
+            mask_1D_tensor = nd.array(mask_1D)
+
+        unked_words = np.where(word_inputs < self._vocab.words_in_train, word_inputs, self._vocab.UNK)
+        word_embs = self.word_embs(nd.array(unked_words, dtype='int'))
+        if self.pret_word_embs:
+            word_embs = word_embs + self.pret_word_embs(nd.array(word_inputs))
+        tag_embs = self.tag_embs(nd.array(tag_inputs))
+
+        # Dropout
+        if is_train:
+            wm, tm = self.generate_emb_mask(seq_len, batch_size)
+            emb_inputs = nd.concat(nd.multiply(wm, word_embs), nd.multiply(tm, tag_embs), dim=2)
+        else:
+            emb_inputs = nd.concat(word_embs, tag_embs, dim=2)  # seq_len x batch_size
+
+        top_recur = biLSTM(self.f_lstm, self.b_lstm, emb_inputs, batch_size,
+                           dropout_x=self.dropout_lstm_input if is_train else 0)
+        if is_train:
+            top_recur = nd.Dropout(data=top_recur, axes=[0], p=self.dropout_mlp)
```

Comment:
Dropout will be automatically turned off in the testing mode so there's no need to add this `if`.